### ISSUE TITLE
AsyncImage caching delete of images crashes app

### ISSUE THREAD

**Issue by ssolari** [kivy/kivy#6103]
### Versions

* Python: 3.7.1
* OS: Mac OS Mojave
* Kivy: 1.10.1
* Kivy installation method: pip

### Description

I am running the app locally, doing nothing more than simply opening the page where a 'close.png' image is used.   It is placed in about 20 boxlayouts within a gridview.   Then just sit and wait and get the error below.

### Code and Logs

```
[INFO   ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "/user/home/demo/main.py", line 287, in <module>
     MainApp().run()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/app.py", line 826, in run
     runTouchApp()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/base.py", line 502, in runTouchApp
     EventLoop.window.mainloop()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/core/window/window_sdl2.py", line 727, in mainloop
     self._mainloop()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/core/window/window_sdl2.py", line 460, in _mainloop
     EventLoop.idle()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/base.py", line 337, in idle
     Clock.tick()
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/clock.py", line 581, in tick
     self._process_events()
   File "kivy/_clock.pyx", line 384, in kivy._clock.CyClockBase._process_events
   File "kivy/_clock.pyx", line 414, in kivy._clock.CyClockBase._process_events
   File "kivy/_clock.pyx", line 412, in kivy._clock.CyClockBase._process_events
   File "kivy/_clock.pyx", line 167, in kivy._clock.ClockEvent.tick
   File "/user/home/anaconda/envs/kivyenv/lib/python3.7/site-packages/kivy/cache.py", line 212, in _purge_by_timeout
     lastaccess = Cache._objects[category][key]['lastaccess']
 KeyError: '<replace_path_to>/image_assets/close.png'
```

**Comment by ssolari** [comment]
Debugging a bit further and tracking the order of events.   The error seems to occur because the ```key``` (path to file) is stored under two different ```category``` names.    The key is removed from the ```kv.texture``` category, after which the error is thrown when attempting to access the same key  from the ```kv.loader``` category.

Are there indirect references between the categories rather than copies?   It seems like this could be the source of the error.

**Comment by tshirtman** [comment]
Can you share your code demonstrating the issue? (if possible simplified as much as possible while keeping the issue), and also, how much time do you typically need to wait? (it might be the cache's timeout time though, the loader and image ones are both at 60s).

**Comment by ssolari** [comment]
@tshirtman It took a surprisingly long time to simplify to the code below.   This was happening after many transitions and loading etc, and I eventually stripped it down to this.   It is a strange bug as shown below.   I have no idea what is causing it, I hope this helps. 

## Code to reproduce
```
from kivy.uix.image import AsyncImage
from kivy.lang.builder import Builder
from kivy.app import App
from kivy.uix.boxlayout import BoxLayout
from kivy.uix.floatlayout import FloatLayout


Builder.load_string('''
<TestApp>:
    # Doing the seemingly exact same thing in kv language below does not produce the same error
    #BoxLayout:
    #    orientation: 'vertical'
    #    AsyncImage:
    #        source: 'loading_icon.gif'
    #    AsyncImage:
    #        source: './close.png'
''')


class TestApp(FloatLayout):

    def __init__(self, **kwargs):
        super(TestApp, self).__init__(**kwargs)
        self.add_widget(AsyncWidgets())


class AsyncWidgets(BoxLayout):

    def __init__(self, **kwargs):
        super(AsyncWidgets, self).__init__(**kwargs)
        self.orientation = 'vertical'
        # replace with './my_loading_icon.gif' and error disappears also
        self.add_widget(AsyncImage(source='loading_icon.gif'))
        self.add_widget(AsyncImage(source='./close.png'))


class MainApp(App):

    def build(self):
        return TestApp()


if __name__ == "__main__":
    MainApp().run()
```
## png used in code
![close](https://user-images.githubusercontent.com/1182784/51502533-4cf07f00-1d8b-11e9-86b7-77899e415c4e.png)

## Odd behaviors

  1) Seemingly doing the exact same thing dirctly in the kv language (uncomment the BoxLayout in and comment out the ```self.add_widget(AsyncWidgets())```)  does not produce the same error.

  2) Using a different gif loading icon makes the error go away.   If instead we use the 'my_loading_icon.gif' below then the error also disappears. 

## other gif that makes error go away
![my_loading_icon](https://user-images.githubusercontent.com/1182784/51502628-c25c4f80-1d8b-11e9-8d62-992f84e38f4a.gif)

**Comment by tito** [comment]
Internally, some Cache are configured to be cleared after a certain time. This is the bug/behavior you're seeing.

**Comment by OutsourcedGuru** [comment]
**Is there any means of configuring/overriding the default Cache behavior in the `~/.kivy/config.ini`, in the `Config.set()` method or similar?**

I'm now getting this same kind of error having upgraded to Raspbian Buster + Kivy 1.11.1 (from Raspbian Stretch + Kivy 1.10.1). It was working as expected before the pair of upgrades. Now it's throwing the following:

```
kivy - WARNING - stderr: Traceback (most recent call last):
kivy - WARNING - stderr:   File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
kivy - WARNING - stderr:     self.run()
kivy - WARNING - stderr:   File "/usr/lib/python2.7/threading.py", line 754, in run
kivy - WARNING - stderr:     self.__target(*self.__args, **self.__kwargs)
kivy - WARNING - stderr:   File "/home/pi/MyPlugin/myplugin/display/__init__.py", line 185, in start
kivy - WARNING - stderr:     OcmLCD.run()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/app.py", line 855, in run
kivy - WARNING - stderr:     runTouchApp()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/base.py", line 504, in runTouchApp
kivy - WARNING - stderr:     EventLoop.window.mainloop()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/core/window/window_egl_rpi.py", line 94, in mainloop
kivy - WARNING - stderr:     self._mainloop()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/core/window/window_egl_rpi.py", line 89, in _mainloop
kivy - WARNING - stderr:     EventLoop.idle()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/base.py", line 339, in idle
kivy - WARNING - stderr:     Clock.tick()
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/clock.py", line 591, in tick
kivy - WARNING - stderr:     self._process_events()
kivy - WARNING - stderr:   File "kivy/_clock.pyx", line 384, in kivy._clock.CyClockBase._process_events
kivy - WARNING - stderr:   File "kivy/_clock.pyx", line 414, in kivy._clock.CyClockBase._process_events
kivy - WARNING - stderr:   File "kivy/_clock.pyx", line 412, in kivy._clock.CyClockBase._process_events
kivy - WARNING - stderr:   File "kivy/_clock.pyx", line 167, in kivy._clock.ClockEvent.tick
kivy - WARNING - stderr:   File "/home/pi/oprint/local/lib/python2.7/site-packages/kivy/cache.py", line 217, in _purge_by_timeout
kivy - WARNING - stderr:     lastaccess = Cache._objects[category][key]['lastaccess']
kivy - WARNING - stderr: KeyError: '/home/pi/.octoprint/data/myplugin/bg/jobreports.png'
kivy - WARNING - stderr: 
```

I'm not bringing in `kivy.cache` anywhere within my plugin. It's on a Raspberry Pi 3B with OctoPrint.

**Comment by OutsourcedGuru** [comment]
Noting this for anyone reading along:

```
<Image,AsyncImage>:
    nocache: True
```

Add this near the top of the `.kv` file to roll the behavior back to what it was before by default. For whatever reasons, upon startup all AsyncImage objects in my code are throwing this on a Raspberry Pi 3B with Raspbian Buster Lite and Kivy 1.11.1. The code worked perfectly before on Raspbian Stretch Lite + Kivy 1.10.1.

**Comment by OutsourcedGuru** [comment]
I'm going to demote my earlier suggestion to just:

```
<AsyncImage>:
    nocache: True
```

I'm not seeing this for standard Image objects, just AsyncImage.

The problem—when seen—varies in impact from just a WARNING in the log to an unresponsive interface to a cycling interface (as driven by `nodm`).

**Comment by tshirtman** [comment]
the merged PR #6815 might fix your issue, if you can test and report back. Thanks.
